https://www.drupal.org/node/1931868 : Fix-rate-widget-display-with-source-translation-on-untranslated-nodes-1931868-1
--- sites/all/modules/contributed/rate/rate.module
+++ sites/all/modules/contributed/rate/rate.module
@@ -438,20 +438,22 @@ function _rate_check_permissions($widget, $node = NULL) {
  * @return int
  */
 function _rate_get_source_translation($entity_type, $entity_id) {
+  $t_entity_id = 0;
   if ($entity_type == 'node' && module_exists('translation')) {
     if (arg(0) == 'node' && arg(1) == $entity_id) {
       // We are on the node page. Use node_load since the node is in static cache.
-      $entity_id = node_load($entity_id)->tnid;
+      $t_entity_id = node_load($entity_id)->tnid;
     }
     else {
       // We are not on the node page. Do not use node_load to prevent executing many useless queries.
       $tnid = db_select('node', 'n')->fields('n', array('tnid'))->condition('n.nid', $entity_id)->execute()->fetchField();
       if ($tnid) {
-        $entity_id = $tnid;
+        $t_entity_id = $tnid;
       }
     }
   }
-  return $entity_id;
+  // Return the original entity_id in case tnid is 0
+  return $t_entity_id ? $t_entity_id : $entity_id;
 }
 
 /**
-- 
1.7.8.msysgit.0

